### DISCUSSION TITLE
ControlNet extension model download wiki page added

### DISCUSSION THREAD

**Discussion by huchenlei** [lllyasviel/stable-diffusion-webui-forge#273] [General]
I found that some users struggles to find download source of ControlNet models.
- https://github.com/lllyasviel/stable-diffusion-webui-forge/issues/226
- https://github.com/lllyasviel/stable-diffusion-webui-forge/issues/259
- https://github.com/lllyasviel/stable-diffusion-webui-forge/discussions/267

I created a wiki page listing all known download sources. Please feel free to add new items if I missed any.
https://github.com/lllyasviel/stable-diffusion-webui-forge/wiki/ControlNet-model-download

**Comment by strawberrymelonpanda** [comment]
Good idea.  Can I suggest adding the SDXL Fooocus inpaint model [now that it works](https://github.com/lllyasviel/stable-diffusion-webui-forge/commit/fb2e2716683453e8b0ba775f7596d8c9b080177d)?

[Model](https://huggingface.co/lllyasviel/fooocus_inpaint/resolve/main/inpaint_v26.fooocus.patch)

This page should probably be linked from the README as well.  I had to Google a few model locations when getting set up.

**Reply by huchenlei** [reply]
Added. Thanks!

**Comment by krummrey** [comment]
Great work for setting the wiki page up. 
New users might find it hard to get ControlNets working. Maybe a page that elaborates a little more beyond just the links might be helpful. I was thinking about something along these lines: (Feel free to correct any mistakes I might have introduced) 

### Introduction
ControlNets allow yout to guide the image generation with user supplied images. 
The original implentation with examples can be found [here](https://github.com/lllyasviel/ControlNet)
### Installation
### Stable Diffusion 1.5

There are three different type of models available of which one needs to be present for ControlNets to function
**LARGE** - these are the original models supplied by the author of ControlNet. Each of them is 1.45 GB large and can be found [here](https://huggingface.co/lllyasviel/ControlNet-v1-1/tree/main).
**MEDIUM** - these have been shrunk to half their size by converting them to 16 bit. Each of them is 723 MB large and can be found [here](https://huggingface.co/comfyanonymous/ControlNet-v1-1_fp16_safetensors/tree/main).
**SMALL** - these are LoRa implementations that only use 136 MB each and can be found [here](https://huggingface.co/comfyanonymous/ControlNet-v1-1_fp16_safetensors/tree/main).

Once downloaded place them in your Forge directory under `models/ControlNet`

### Stable Diffusion XL

The original XL ControlNet models can be found [here](https://huggingface.co/lllyasviel/sd_control_collection/tree/main). They too come in three sizes from small to large.
Alternative models have been released here (Link seems to direct to SD1.5 models)

After download the models need to be placed in the same directory as for 1.5 `models/ControlNet`

### Extensions

Other projects have adapted the ControlNet method and have released their models:

**Animal Openpose** [Original Project repo](https://github.com/Mikubill/sd-webui-controlnet/discussions/2351) - [Models](https://huggingface.co/huchenlei/animal_openpose/tree/main)

**IPAdapter** [Original Project repo](https://github.com/tencent-ailab/IP-Adapter) - [Models](https://huggingface.co/h94/IP-Adapter/tree/main)

**IPAdapter FaceID** [Original Project repo](https://huggingface.co/h94/IP-Adapter-FaceID) - [Models](https://huggingface.co/h94/IP-Adapter-FaceID/tree/main)

**PhotoMaker** [Original Project repo](https://github.com/TencentARC/PhotoMaker) - [Models](https://github.com/TencentARC/PhotoMaker#-download-models)

**InstantID** [Original Project repo](https://github.com/InstantID/InstantID) - Follow instruction in [here](https://github.com/Mikubill/sd-webui-controlnet/discussions/2589) - Needs a little more explanation

**Fooocus Inpaint**  [patch](https://huggingface.co/lllyasviel/fooocus_inpaint/resolve/main/inpaint_v26.fooocus.patch) - Needs a little more explanation

### T2I Adapter

Are they still a thing? https://huggingface.co/TencentARC/T2I-Adapter/tree/main/models

**Reply by huchenlei** [reply]
Thanks! I have updated the wiki page.

**Comment by xhoxye** [comment]
SVD, Z123 model download address can also be listed?

**Comment by aidenli** [comment]
Thanks, it saved me a lot of time looking for these models!

**Comment by BenDes21** [comment]
Hi, is it possible to install Hand Refiner also ?
